src/coreclr/debug/daccess/request.cpp

```diff
@@ -3551,6 +3553,118 @@ ClrDataAccess::TraverseVirtCallStubHeap(CLRDATA_ADDRESS pAppDomain, VCSHeapType
     return hr;
 }
 
+HRESULT ClrDataAccess::GetDomainLoaderAllocator(CLRDATA_ADDRESS domainAddress, CLRDATA_ADDRESS *pLoaderAllocator)
+{
+    if (pLoaderAllocator == nullptr)
+        return E_INVALIDARG;
+
+    if (domainAddress == 0)
+    {
+        *pLoaderAllocator = 0;
+        return S_FALSE;
+    }
+
+    SOSDacEnter();
+
+    PTR_BaseDomain pDomain = PTR_BaseDomain(TO_TADDR(domainAddress));
+    *pLoaderAllocator = pDomain != nullptr ? HOST_CDADDR(pDomain->GetLoaderAllocator()) : 0;
+
+    SOSDacLeave();
+    return hr;
+}
+
+// The ordering of these entries must match the order enumerated in GetLoaderAllocatorHeaps
+static const char *LoaderAllocatorLoaderHeapNames[] = 
+{
+    "LowFrequencyHeap",
+    "HighFrequencyHeap",
+    "StubHeap",
+    "ExecutableHeap",
+    "FixupPrecodeHeap",
+    "NewStubPrecodeHeap",
+    "IndcellHeap",
+    "LookupHeap",
+    "ResolveHeap",
+    "DispatchHeap",
+    "CacheEntryHeap",
+    "VtableHeap",
+};
+
+const int LoaderHeapCount = 12;
```

Comment:
I don't understand the details of the loader heap, but is hard coding the number of heaps the only way? What if it changes?  Is there any static_assert or something to at least make devs aware of this?

Reply:
The value itself isn't fixed.  We can add more entries to `LoaderAllocatorLoaderHeapNames` by increasing `LoaderHeapCount` and adding more (or less) heaps.  It's actually only determining the length of the above array.  I really want to use `_countof(LoaderAllocatorLoaderHeapNames)` below, but `_countof` is only available in MSVC compilers, so instead we have to make a constant to use throughout the code that needs to know the length of that array.


I could make this more obvious by using the value like this:

``` C++
const int LoaderHeapCount = 12;
static const char *LoaderAllocatorLoaderHeapNames[LoaderHeapCount] = 
```

I'll push that change.

Reply:
See: https://github.com/dotnet/runtime/pull/82437/commits/301089ce7dec09e9233592ed3f19f5bf322e1fc8.

Reply:
Looks good.

Reply:
Could this be sizeof(LoaderAllocatorLoaderHeapNames)/sizeof(const char *)?

Reply:
It's now `ARRAY_SIZE`, I couldn't seem to find that macro when I was using `_countof`.